packages/components/popover/src/PopoverClose.tsx

```diff
@@ -0,0 +1,42 @@
+import * as RadixPopover from '@radix-ui/react-popover'
+import { Icon } from '@spark-ui/icon'
+import { IconButton } from '@spark-ui/icon-button'
+import { Close as CloseSVG } from '@spark-ui/icons/dist/icons/Close'
+import { cx } from 'class-variance-authority'
+import { forwardRef, useEffect } from 'react'
+
+import { usePopover } from './PopoverContext'
+
+export type CloseProps = RadixPopover.PopoverCloseProps & {
+  'aria-label': string
+}
+
+export const Close = forwardRef<HTMLButtonElement, CloseProps>(
+  ({ 'aria-label': ariaLabel, className, ...rest }, ref) => {
+    const { setHasCloseButton } = usePopover()
+
+    useEffect(() => {
+      setHasCloseButton(true)
```

Comment:
Isn't this going to cause a layout shift ?

Reply:
I think that with `useLayoutEffect` the problem is going to still happen but we can review together later on.